solidity/random-beacon/test/ReimbursmentPool.test.ts

```diff
@@ -0,0 +1,505 @@
+
+          const refundeeBalanceBefore = await provider.getBalance(
+            refundee.address
+          )
+
+          await reimbursementPool
+            .connect(thirdPartyContract)
+            .refund(50000, refundee.address)
+
+          const refundeeBalanceAfter = await provider.getBalance(
+            refundee.address
+          )
+          const refundeeBalanceDiff = refundeeBalanceAfter.sub(
+            refundeeBalanceBefore
+          )
+          // gas spent + static gas => 50k + 37.5k
+          expect(refundeeBalanceDiff).to.be.eq(
+            ethers.utils.parseUnits("87500", "gwei")
+          )
+        })
+      })
+
+      context("when receiver address is zero", () => {
+        it("should revert", async () => {
+          await expect(
+            reimbursementPool
+              .connect(thirdPartyContract)
+              .refund(50000, ZERO_ADDRESS)
+          ).to.be.revertedWith("Receiver's address cannot be zero")
+        })
+      })
+
+      context("when no funds available in the pool", () => {
+        let tx: Promise<ContractTransaction>
+
+        beforeEach(async () => {
+          await createSnapshot()
+
+          await reimbursementPool
+            .connect(owner)
+            .setMaxGasPrice(ethers.utils.parseUnits("1.0", "gwei"))
+
+          await reimbursementPool.connect(owner).withdrawAll(thirdParty.address)
+
+          tx = reimbursementPool
+            .connect(thirdPartyContract)
+            .refund(50000, refundee.address)
+        })
+
+        afterEach(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should not revert", async () => {
+          await expect(tx).to.not.be.reverted
+        })
+
+        it("should emit SendingEtherFailed event", async () => {
+          // gas spent + static gas => 50k + 37.5k
+          await expect(tx)
+            .to.emit(reimbursementPool, "SendingEtherFailed")
+            .withArgs(
+              ethers.utils.parseUnits("87500", "gwei"),
+              refundee.address
+            )
+        })
+      })
+    })
+  })
+
+  describe("authorize", () => {
+    context("when the caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          reimbursementPool
+            .connect(thirdParty)
+            .authorize(thirdPartyContract.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the caller is the owner", () => {
+      it("should authorize a contract", async () => {
+        const tx = await reimbursementPool
+          .connect(owner)
+          .authorize(thirdPartyContract.address)
+
+        expect(await reimbursementPool.isAuthorized(thirdPartyContract.address))
+          .to.be.true
+
+        await expect(tx)
+          .to.emit(reimbursementPool, "thirdPartyContract")
+          .withArgs(thirdPartyContract.address)
+      })
+    })
+  })
+
+  describe("unuthorize", () => {
```

Comment:
We should have a before block where we first authorize the address.

Reply:
good point [c03c038](https://github.com/keep-network/keep-core/pull/2826/commits/c03c038461a772771d7ae100b372f9c81d46c230)